### PR TITLE
Fix m vector download

### PR DESCRIPTION
**Pull request by achang28**
*For review, at this point!*  Use the following parameters to test Prod vs. this fix:
Cohort:  **TCGA Esophageal Cancer**
Dataset:  **somatic mutation SNPs and small INDELs (ucsc automated vcf)**
Gene:  **TP53**

Note:  I manually changed the following attached files to .txt so that Github their upload.
Issue:  https://github.com/ucscXena/ucsc-xena-client/issues/52

Prod Results (unfiltered):
[MutationVector-Prod.txt](https://github.com/ucscXena/ucsc-xena-client/files/162373/MutationVector-Prod.txt)

Dev Results:
[MutationVector-Dev.txt](https://github.com/ucscXena/ucsc-xena-client/files/162370/MutationVector-Dev.txt)

**Conditions Covered**:
1) Empty rows (null or array of 0):  **done**
2) Sample Variations: **done**
3) Samples w/ blank mutations: **done**
4) Samples with no mutations:  **done**
5) Samples with mutations:  **done**

### CODE REVIEW

File: js/plotMutationVector.js [hunk 1]
```diff
@@ -211,6 +211,21 @@ var MutationColumn = hotOrNot(React.createClass({
 	componentWillUnmount: function () {
 		this.ttevents.dispose();
 	},
+	onDownload: function() {
+		let {column, samples, data, index} = this.props,
+			rows = _.getIn(data, ['req', 'rows']),
+			fields = _.keys(_.getIn(index, ['byPosition', 'el'])),
```
Comment (acthp) [inline js/plotMutationVector.js]: Get field list from ```rows[0]```, not ```byPosition```. Treat ```byPosition``` as opaque. It should really only be accessed via the interval-tree methods. Be sure to handle the case when ```rows``` is empty.

**Comment by achang28** [comment]
okay -- will do on both.  Thanks for clarification.

**Comment by achang28** [comment]
Remove the following comments?
`// XXX
//	if (this.ws.column.dataType === 'clinicalMatrix') {
//		fieldNames = ['sample'].concat([this.ws.column.fieldLabel.default]);
//	}`

**Comment by acthp** [comment]
The comment can be removed. It was left there because it's unimplemented vs. production. I'll file an issue.

**Comment by achang28** [comment]
Jing explained in greater detail the context behind TSV downloads of Mutation Vector samples from Prod.  Given that these downloads have corresponding samples on the same row by design, and that some of the samples with `no mutation` are also listed within these downloads, could `this.props.index.bySample` be useful?  If reliable, it could be useful given how each object holds the instances of each sample -- 0 or many...

this props index bysample

**Comment by acthp** [comment]
Yes, it's just ```groupBy(rows, 'sample')```, so you can map the list of samples over it to get rows for each sample.

**Comment by achang28** [comment]
Closing PR due to Dev being merged with Prod

**Pull request closed without merging**